sqlfs/writer.go

```diff
@@ -2,51 +2,46 @@ package sqlfs
 
 import (
 	"database/sql"
+	"encoding/base64"
 	"fmt"
 )
 
-var bufSize = 4 * 1024
+const bufSize = 4 * 1024
 
 // Writer implements io.WriteCloser.
 type Writer struct {
-	db     *sql.DB
-	table  string
-	buf    []byte
-	insert *sql.Stmt
+	db      *sql.DB
+	table   string
+	buf     []byte
+	flushID int
 }
 
 // Create creates a new table or truncates an existing table and
 // returns a writer.
-func Create(db *sql.DB, table string) (*Writer, error) {
+func Create(db *sql.DB, driver, table string) (*Writer, error) {
 	if e := dropTable(db, table); e != nil {
 		return nil, fmt.Errorf("create: %v", e)
 	}
-	return Append(db, table)
-}
-
-// Append returns a writer to append to an existing table.  It creates
-// the table if it doesn't exist.
-func Append(db *sql.DB, table string) (*Writer, error) {
-	if e := createTable(db, table); e != nil {
+	if e := createTable(db, driver, table); e != nil {
 		return nil, fmt.Errorf("create: %v", e)
 	}
-	return &Writer{db, table, make([]byte, 0, bufSize), nil}, nil
+	return &Writer{db, table, make([]byte, 0, bufSize), 0}, nil
 }
 
 // Write write bytes to sqlfs and returns (num_bytes, error)
 func (w *Writer) Write(p []byte) (n int, e error) {
 	n = 0
-	for len(p) > 0 {
+	for nBytes := len(p); nBytes > 0; nBytes = len(p) {
 		fill := bufSize - len(w.buf)
-		if fill > len(p) {
-			fill = len(p)
+		if fill > nBytes {
+			fill = nBytes
 		}
 		w.buf = append(w.buf, p[:fill]...)
 		p = p[fill:]
 		n += fill
 		if len(w.buf) >= bufSize {
 			if e = w.flush(); e != nil {
-				return n, fmt.Errorf("writer flush failed: %v", e)
```

Comment:
This might be a leftover for debugging. Could you change back to `return n, fmt.Errorf("writer flush failed: %v", e)`?